OrbitVulkanLayer/SubmissionTrackerTest.cpp

```diff
@@ -0,0 +1,1410 @@
+      [](VkDevice /*device*/, uint32_t* allocated_slot) -> bool {
+    *allocated_slot = kSlotIndex3;
+    return true;
+  };
+
+  const std::function<bool(VkDevice, uint32_t*)> mock_next_ready_query_slot_function_4 =
+      [](VkDevice /*device*/, uint32_t* allocated_slot) -> bool {
+    *allocated_slot = kSlotIndex4;
+    return true;
+  };
+
+  const std::function<bool(VkDevice, uint32_t*)> mock_next_ready_query_slot_function_5 =
+      [](VkDevice /*device*/, uint32_t* allocated_slot) -> bool {
+    *allocated_slot = kSlotIndex5;
+    return true;
+  };
+
+  const std::function<bool(VkDevice, uint32_t*)> mock_next_ready_query_slot_function_6 =
+      [](VkDevice /*device*/, uint32_t* allocated_slot) -> bool {
+    *allocated_slot = kSlotIndex6;
+    return true;
+  };
+
+  const std::function<bool(VkDevice, uint32_t*)> mock_next_ready_query_slot_function_7 =
+      [](VkDevice /*device*/, uint32_t* allocated_slot) -> bool {
+    *allocated_slot = kSlotIndex7;
+    return true;
+  };
+
+  static constexpr uint64_t kTimestamp1 = 11;
+  static constexpr uint64_t kTimestamp2 = 12;
+  static constexpr uint64_t kTimestamp3 = 13;
+  static constexpr uint64_t kTimestamp4 = 14;
+  static constexpr uint64_t kTimestamp5 = 15;
+  static constexpr uint64_t kTimestamp6 = 16;
+  static constexpr uint64_t kTimestamp7 = 17;
+
+  const PFN_vkGetQueryPoolResults mock_get_query_pool_results_function_all_ready =
+      +[](VkDevice /*device*/, VkQueryPool /*queryPool*/, uint32_t first_query,
+          uint32_t query_count, size_t /*dataSize*/, void* data, VkDeviceSize /*stride*/,
+          VkQueryResultFlags flags) -> VkResult {
+    EXPECT_EQ(query_count, 1);
+    EXPECT_NE((flags & VK_QUERY_RESULT_64_BIT), 0);
+    switch (first_query) {
+      case kSlotIndex1:
+        *absl::bit_cast<uint64_t*>(data) = kTimestamp1;
+        break;
+      case kSlotIndex2:
+        *absl::bit_cast<uint64_t*>(data) = kTimestamp2;
+        break;
+      case kSlotIndex3:
+        *absl::bit_cast<uint64_t*>(data) = kTimestamp3;
+        break;
+      case kSlotIndex4:
+        *absl::bit_cast<uint64_t*>(data) = kTimestamp4;
+        break;
+      case kSlotIndex5:
+        *absl::bit_cast<uint64_t*>(data) = kTimestamp5;
+        break;
+      case kSlotIndex6:
+        *absl::bit_cast<uint64_t*>(data) = kTimestamp6;
+        break;
+      case kSlotIndex7:
+        *absl::bit_cast<uint64_t*>(data) = kTimestamp7;
+        break;
+      default:
+        UNREACHABLE();
+    }
+    return VK_SUCCESS;
+  };
+
+  const PFN_vkGetQueryPoolResults mock_get_query_pool_results_function_not_ready =
+      +[](VkDevice /*device*/, VkQueryPool /*queryPool*/, uint32_t /*first_query*/,
+          uint32_t /*query_count*/, size_t /*dataSize*/, void* /*data*/, VkDeviceSize /*stride*/,
+          VkQueryResultFlags /*flags*/) -> VkResult { return VK_NOT_READY; };
+
+  void EXPECT_SINGLE_COMMAND_BUFFER_SUBMISSION_EQ(
```

Comment:
I don't think these should be in all caps, even if they resemble `EXPECT_EQ`, etc.

Reply:
Also do these functions need to be in the class? Cant' they just be in the anonoymous namespace?

Reply:
Now all related information (the mock methods, the indices and the timestamps are very close together, which I kind of like.

Reply:
Yeah, make sense, I don't know it it's ideal but I'm totally fine with it.